scala/private/phases/phase_compile.bzl

```diff
@@ -480,17 +480,20 @@ def _try_to_compile_java_jar(
         java_compilation_provider = provider,
     )
 
-def _adjust_resources_path(path, resource_strip_prefix):
+def _adjust_resources_path(resource, resource_strip_prefix):
+    path = resource.path
     if resource_strip_prefix:
-        return _adjust_resources_path_by_strip_prefix(path, resource_strip_prefix)
+        root = resource.owner.workspace_root + "/" if (resource.owner.workspace_root) else ""
```

Comment:
maybe this logic should be in `_adjust_resources_path_by_strip_prefix`? Make it take a `resource`
I'm also bummed about having this logic here since it's very common starlark need and I have a hunch it exists already in bazel--skylib but I won't ask you now to go down that route